--- a/qutebrowser/components/adblock.py
+++ b/qutebrowser/components/adblock.py
@@ -119,6 +119,14 @@ class HostBlocker:
             return False
 
         host = request_url.host()
+        path = request_url.path()
+        query = request_url.query()
+
+        if (host == 'www.youtube.com' and
+          ((path == '/get_video_info' and '&adformat=' in query) or
+           (path == '/get_midroll_info'))):
+            return True
+
         return ((host in self._blocked_hosts or
                  host in self._config_blocked_hosts) and
                 not _is_whitelisted_url(request_url))
