diff --git a/Makefile b/../Makefile
index 120ce25..aefb192 100644
--- a/Makefile
+++ b/Makefile
@@ -6,7 +6,7 @@ include config.mk
 
 REQ = util
 HDR = arg.h
-BIN = png2ff ff2png jpg2ff ff2jpg ff2pam ff2ppm
+BIN = png2ff ff2png ff2pam ff2ppm
 SCR = 2ff
 MAN1 = 2ff.1 $(BIN:=.1)
 MAN5 = farbfeld.5
@@ -15,20 +15,20 @@ all: $(BIN)
 
 png2ff-LDLIBS = $(PNG-LDLIBS)
 ff2png-LDLIBS = $(PNG-LDLIBS)
-jpg2ff-LDLIBS = $(JPG-LDLIBS)
-ff2jpg-LDLIBS = $(JPG-LDLIBS)
+#jpg2ff-LDLIBS = $(JPG-LDLIBS)
+#ff2jpg-LDLIBS = $(JPG-LDLIBS)
 
 png2ff: png2ff.o $(REQ:=.o)
 ff2png: ff2png.o $(REQ:=.o)
-jpg2ff: jpg2ff.o $(REQ:=.o)
-ff2jpg: ff2jpg.o $(REQ:=.o)
+#jpg2ff: jpg2ff.o $(REQ:=.o)
+#ff2jpg: ff2jpg.o $(REQ:=.o)
 ff2pam: ff2pam.o $(REQ:=.o)
 ff2ppm: ff2ppm.o $(REQ:=.o)
 
 png2ff.o: png2ff.c config.mk $(HDR) $(REQ:=.h)
 ff2png.o: ff2png.c config.mk $(HDR) $(REQ:=.h)
-jpg2ff.o: jpg2ff.c config.mk $(HDR) $(REQ:=.h)
-ff2jpg.o: ff2jpg.c config.mk $(HDR) $(REQ:=.h)
+#jpg2ff.o: jpg2ff.c config.mk $(HDR) $(REQ:=.h)
+#ff2jpg.o: ff2jpg.c config.mk $(HDR) $(REQ:=.h)
 ff2pam.o: ff2pam.c config.mk $(HDR) $(REQ:=.h)
 ff2ppm.o: ff2ppm.c config.mk $(HDR) $(REQ:=.h)
 
