diff --git a/hsetroot.c b/hsetroot.c
index a6bd1f3..4d82e4e 100644
--- a/hsetroot.c
+++ b/hsetroot.c
@@ -32,7 +32,7 @@ usage(char *commandline)
     "\n"
     "Image files:\n"
     " -center <image>            Render an image centered on screen\n"
-    " -cover <image>             Render an image centered on screen scaled to fill the screen fully\n"
+    " -scale <image>             Render an image centered on screen scaled to fill the screen fully\n"
     " -tile <image>              Render an image tiled\n"
     " -full <image>              Render an image maximum aspect\n"
     " -extend <image>            Render an image max aspect and fill borders\n"
@@ -450,7 +450,7 @@ main(int argc, char **argv)
           fprintf (stderr, "Bad image (%s)\n", argv[i]);
           continue;
         }
-      } else if (strcmp(argv[i], "-cover") == 0) {
+      } else if (strcmp(argv[i], "-scale") == 0) {
         if ((++i) >= argc) {
           fprintf(stderr, "Missing image\n");
           continue;
