diff --git a/bgs.c b/bgs.c
index 90cf702..8c4eed5 100644
--- a/bgs.c
+++ b/bgs.c
@@ -89,15 +89,17 @@ drawbg(void) {
 			ny = monitors[i].y + (monitors[i].h - nh) / 2;
 			break;
 		case ModeZoom:
-			nw = monitors[i].w;
-			nh = monitors[i].h;
-			if(w > h && (w / h > (monitors[i].w / monitors[i].h))) {
+			if(((float)w / h) > ((float)monitors[i].w / monitors[i].h)) {
+				nh = monitors[i].h;
+				nw = nh * w / h;
 				nx = monitors[i].x + (monitors[i].w - nw) / 2;
-				ny = monitors[i].y + (int)ceil(h * nx / w) / 2;
+				ny = monitors[i].y;
 			}
 			else {
+				nw = monitors[i].w;
+				nh = nw * h / w;
 				ny = monitors[i].y + (monitors[i].h - nh) / 2;
-				nx = monitors[i].x + (int)ceil(w * ny / h) / 2;
+				nx = monitors[i].x;
 			}
 			break;
 		default: /* ModeScale */
